### PR TITLE
595 lock down authentication

### LINKED ISSUE #595: Lock down application for unauthenticated users

### PR DESCRIPTION
**Pull request by roschaefer**
close #595

### CODE REVIEW

File: spec/views/layouts/application.html.haml_spec.rb [hunk 1]
```diff
@@ -1,85 +1,101 @@
...
-    end
+      it 'has multiple levels' do
+        expect(parsed).to have_css('.subnav__select', count: 2)
+      end
 
-    it 'has multiple select options with respective labels and urls' do
-      expect(@parsed).to have_css('.subnav__breadcrumb__item:nth-child(1) select option', count: 3)
-      expect(@parsed).to have_css('.subnav__breadcrumb__item:nth-child(2) select option', count: 2)
+      it 'has multiple select options with respective labels and urls' do
+        expect(parsed).to have_css('.subnav__breadcrumb__item:nth-child(2) select option', count: 3)
```
Comment (roschaefer) [inline spec/views/layouts/application.html.haml_spec.rb]: hey @tillprochaska can you have a look into this before we merge it? Why do I have to increment the `nth-child` here?

Comment (roschaefer) [inline spec/views/layouts/application.html.haml_spec.rb]: When I debug it I get this:
```
[6] pry(#<RSpec::ExampleGroups::LayoutsApplication::WhenLoggedIn::RendersABreadcrumbMenu>)> parsed.find('.subnav__breadcrumb__item:nth-child(1)', visible: false)
Capybara::ElementNotFound: Unable to find css ".subnav__breadcrumb__item:nth-child(1)"
from /home/robert/.gem/ruby/2.4.1/gems/capybara-2.15.1/lib/capybara/node/finders.rb:313:in `block in synced_resolve'
```

Huh?

Comment (drjakob) [inline spec/views/layouts/application.html.haml_spec.rb]: Huh?

Comment (tillprochaska) [reply spec/views/layouts/application.html.haml_spec.rb]: @roschaefer: I did not take a closer look at this. but in #623 I had to decrement the index again to make tests work. Either I completely overlooked something or this is an unintended behavior of PhantomJS…

**Pull request merged**